pkg/storage/chunk/gcp/gcs_object_client.go

```diff
@@ -84,7 +84,12 @@ func newGCSObjectClient(ctx context.Context, cfg GCSConfig, hedgingCfg hedging.C
 
 func newBucketHandle(ctx context.Context, cfg GCSConfig, hedgingCfg hedging.Config, hedging bool, clientFactory ClientFactory) (*storage.BucketHandle, error) {
 	var opts []option.ClientOption
-	httpClient, err := gcsInstrumentation(ctx, storage.ScopeReadWrite, cfg.Insecure, cfg.EnableHTTP2)
+	enableHTTP2 := cfg.EnableHTTP2
+	if !hedging {
+		// enforce http2 for non-hedging bucket (PUT/LIST/DELETE)
```

Comment:
I'm a little nervous about this; there's nothing explicit disabling this for non-`GET` requests right now - except that we only hedge `GET` requests (right now).

Don't let this block you, but I'd hope we could make this a little stronger. This is how nasty bugs slip in IMHO.

Reply:
yeah I can find a way to refactor this to make more clear. I went for the easier route.

Reply:
I'm not totally sure why we need this? the default is for hedging to be disabled and http2 to be enabled.

I'm not sure what scenario this protects against, what would be the problem of disabling http2 if someone wanted to and they were also not using hedging?

Reply:
Shouldn't hedging and http2 be independent options?

Reply:
I'm trying to disable http2 only for Gets the code is clunky but does it. I'll refactor it to make it clear